axlearn/cloud/gcp/job.py

```diff
@@ -639,6 +639,384 @@ def _execute(self) -> Any:
         )
 
 
+class GPUGKEJob(GKEJob):
+    """A GPU job represented as a k8s JobSet.
+
+    See also `gke_runner` as an example.
+    """
+
+    @config_class
+    class Config(GKEJob.Config):
+        """Configures GPUGKEJob.
+
+        Attributes:
+            accelerator: GPU configuration.
+            queue: The Kueue LocalQueue to use. If not set, no queue is used.
+        """
+
+        accelerator: AcceleratorConfig = AcceleratorConfig()
+        queue: Optional[str] = None
+
+    @classmethod
+    def define_flags(cls, fv: flags.FlagValues):
+        super().define_flags(fv)
+        common_kwargs = dict(flag_values=fv, allow_override=True)
+        accelerator_flags(**common_kwargs)
+        flags.DEFINE_string(
+            "queue",
+            None,
+            "The name of the Kueue LocalQueue to use. If not set, no queue is used.",
+            **common_kwargs,
+        )
+
+    @classmethod
+    def from_flags(cls, fv: flags.FlagValues, **kwargs) -> Config:
+        cfg: GPUGKEJob.Config = super().from_flags(fv, **kwargs)
+        cfg.accelerator.set(instance_type=fv.instance_type, num_replicas=fv.num_replicas)
+        return cfg
+
+    def __init__(self, cfg: Config):
+        bundler_cfg = cfg.bundler
+        bundler_cfg = getattr(bundler_cfg, "inner", bundler_cfg)
+        if bundler_cfg is None or not issubclass(bundler_cfg.klass, BaseDockerBundler):
+            raise NotImplementedError(f"Only docker bundler supported, got: {bundler_cfg}")
+        super().__init__(cfg)
+        if cfg.gcsfuse_mount:
+            raise NotImplementedError("GCSFuse is not supported on GKE with GPU.")
+        if cfg.enable_pre_provisioner:
+            raise NotImplementedError("Pre-provisioner is not supported on GKE with GPU.")
+        instance_type = cfg.accelerator.instance_type
+        if not instance_type.startswith("gpu-a3-highgpu"):
+            raise NotImplementedError(
+                f"The instance type {instance_type} is not supported on GKE with GPU."
+                "Only gpu-a3-highgpu-8g is supported"
```

Comment:
I do plan to have a follow up PR for a3-megagpu very soon. That being said, to keep code simple for today's needs, it may be better to remove all the if checks and come up with a better mechanism once we support multiple types.

Reply:
fixed

Reply:
Should we fix the formatting?

Reply:
pushed and closing